STATE_TRANSITIONS.MD

```diff
@@ -0,0 +1,224 @@
+# Run State Transitions
+Run states change based on the success or failure of a job run. The datasets consumed and/or produced
+by a job run are immutable and do not change based on the success or failure of the run.
+
+# Dataset State Transitions
+Dataset states change based on the success or failure of job runs that consume or generate those 
+datasets. Typically, Runs generate dataset versions as outputs, but sometimes a job run may result 
+in an input dataset version being created if that dataset did not previously exist in the database.
+Once a dataset exists, a new version will only be created if the dataset is the output of a job run. 
+
+### Base case
+* Given
+  * JobA does not exist
+  * DatasetX does not exist
+  * DatasetY does not exist
+* When
+  * JobA@Run1 **succeeds**
+  * JobA consumes DatasetX
+  * JobA produces DatasetY
+* Then
+  * Create JobA V1
+  * Create DatasetX@V1
+  * Set DatasetX current version to V1
+  * JobA@Run1 consumes DatasetX@V1
+  * Create DatasetY@V1
+  * Set DatasetY current version to V1
+  * JobA@Run1 produces DatasetY@V1
+
+### Consume existing dataset
+* Given
+    * JobA does not exist
+    * DatasetX@V1 exists
+    * DatasetY does not exist
+* When
+    * JobA@Run1 **succeeds**
+    * JobA consumes DatasetX
+    * JobA produces DatasetY
+* Then
+    * Create JobA@Run1
+    * JobA@Run1 consumes DatasetX@V1
+    * Create DatasetY@V1
+    * Set DatasetY current version to V1
+    * JobA@Run1 produces DatasetY@V1
+
+### Consume and produce existing datasets
+* Given
+    * JobA exists
+    * DatasetX@V1 exists
+    * DatasetY@V1 exists
+* When
+    * JobA@Run2 **succeeds**
+    * JobA consumes DatasetX
+    * JobA produces DatasetY
+* Then
+    * Create JobA@Run2
+    * JobA@Run1 consumes DatasetX@V1
+    * Create DatasetY@V2
+    * Set DatasetY current version to V2
+    * JobA@Run2 produces DatasetY@V2
+
+### Successful job chain
+* Given
+    * JobA does not exist
+    * JobB does not exist
+    * DatasetX does not exist
+    * DatasetY does not exist
+* When
+    * JobA@Run1 **succeeds**
+    * JobA produces DatasetX
+    * JobB consumes DatasetX
+    * JobB produces DatasetY
```

Comment:
In previous cases we have `DS X -> Job A -> DS Y`
I would assume we want to keep the same shape in all cases.
So the case should be `DS X -> Job A -> DS Y -> Job B -> DS Z`, correct?

Reply:
summarizing this at the top would help